AmplifyPlugins/Predictions/AWSPredictionsPlugin/Support/Utils/IdentifyTextResultTransformers.swift

```diff
@@ -137,141 +111,269 @@ class IdentifyTextResultTransformers: IdentifyResultTransformers {
+                let text = keyBlock.text,
+                case .word = keyBlock.blockType else {
+                continue
+            }
+            keyText += text + " "
+        }
+        return keyText.trimmingCharacters(in: .whitespacesAndNewlines)
+    }
+
+    static func processValueOfKeyValueSet(ids: [String],
+                                          blockMap: [String: AWSTextractBlock]) -> (String, Bool) {
+        var valueText = ""
+        var isSelected = false
+        var selectionItemFound = false
+
+        for valueId in ids {
+            guard let valueBlock = blockMap[valueId],
+                let valueBlockRelations = valueBlock.relationships else {
+                continue
+            }
+
+            for valueBlockRelation in valueBlockRelations {
+                guard let wordBlockIds = valueBlockRelation.ids else {
+                    break
+                }
+
+                for wordBlockId in wordBlockIds {
+                    guard let wordBlock = blockMap[wordBlockId] else {
                         continue
                     }
-                    switch valueBlockType {
-                    case .word:
+                    let wordValueBlockType = wordBlock.blockType
+                    let selectionStatus = wordBlock.selectionStatus
 
-                        if let text = valueBlock?.text {
-                        valueText += text + " "
+                    switch wordValueBlockType {
+                    case .word:
+                        if let text = wordBlock.text {
+                            valueText += text + " "
                         }
                     case .selectionElement:
-                        valueSelected = keyBlock.selectionStatus == .selected ? true : false
+                        let result = processSelectionElement(selectionItemFound, selectionStatus: selectionStatus)
+                        isSelected = result.0
+                        selectionItemFound = result.1
                     default: break
                     }
                 }
-
-            default:
-                break
             }
         }
+        return (valueText.trimmingCharacters(in: .whitespacesAndNewlines), isSelected)
+    }
 
-        guard let boundingBox = processBoundingBox(keyBlock.geometry?.boundingBox) else {
-            return nil
+    static func processSelectionElement(_ selectionItemFoundOld: Bool,
+                                        selectionStatus: AWSTextractSelectionStatus) -> (Bool, Bool) {
+        var selectionItemFound = selectionItemFoundOld
+        var isSelected = false
+        if !selectionItemFound {
+            selectionItemFound = true
+            //TODO: Support multiple selection items found in a single cell
+            isSelected = selectionStatus == .selected
+        } else {
+            Amplify.log.error("Multiple selection items found in single cell")
         }
+        return (isSelected, selectionItemFound)
+    }
```

Comment:
I understand trying to re-use code here, but in this case, I think it actually adds complexity.  Lets remove this function.